modules/ding_nodelist/ding_nodelist.admin.inc

```diff
@@ -0,0 +1,186 @@
+    '#type' => 'fieldset',
+    '#title' => t('Templates'),
+    '#tree' => TRUE,
+    '#theme' => '_ding_nodelist_templates_table',
+  );
+
+  $templates = _ding_nodelist_get_templates();
+
+  foreach ($templates as $template_name => $template) {
+    $form['templates'][$template_name]['template_name'] = array(
+      '#type' => 'hidden',
+      '#value' => $template_name,
+    );
+
+    $form['templates'][$template_name]['title'] = array(
+      '#type' => 'textfield',
+      '#default_value' => isset($template['title']) ? $template['title'] : '',
+      '#size' => 40,
+    );
+
+    $ct = node_type_get_names();
+    $form['templates'][$template_name]['content_type'] = array(
+      '#markup' => isset($ct[$template['content_type']]) ?
+      $ct[$template['content_type']] : t('Missing (@ctname)', array('@ctname' => $template['content_type'])),
+    );
+
+    $widgets = _ding_nodelist_get_widget_types();
+    $form['templates'][$template_name]['widget'] = array(
+      '#type' => 'select',
+      '#empty_option' => t('- select -'),
+      '#required' => FALSE,
+      '#default_value' => isset($template['widget']) ? $template['widget'] : '',
+      '#options' => $widgets,
+    );
+
+    // Visible in list of templates?
+    $form['templates'][$template_name]['visible'] = array(
+      '#type' => 'checkbox',
+      '#default_value' => $template['status'] == NL_TPL_ACTIVE ? TRUE : FALSE,
+    );
+  }
+
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save'),
+  );
+
+  return $form;
+}
+
+/**
+ * Writes details about found templates.
+ */
+function ding_nodelist_templates_form_submit($form, &$form_state) {
+  foreach ($form_state['values']['templates'] as $template) {
+    $tpl_record = array(
+      'title' => $template['title'],
+      'status' => $template['visible'] ? NL_TPL_ACTIVE : NL_TPL_HIDDEN,
+      'widget' => $template['widget'],
+    );
+    db_update('ding_nodelist_templates')->fields($tpl_record)
+      ->condition('filename', $template['template_name'])->execute();
+  }
+
+  drupal_set_message(t('Template settings have been saved.'));
+}
+
+/**
+ * Trim content settings form.
+ */
+function ding_nodelist_trim_content($form, &$form_state) {
+  $form['trim_content'] = array(
+    '#type' => 'fieldset',
+    '#collapsible' => FALSE,
+    '#title' => t('Text truncation in widgets'),
+    '#description' => t('Define amount of chars to be shown in each widget type'),
+    '#theme' => '_ding_nodelist_trim_text_table',
+    '#tree' => TRUE,
+  );
+  $widgets = _ding_nodelist_widgets_content_trim();
+  foreach ($widgets as $widget) {
+    $form['trim_content'][$widget]['widget'] = array(
+      '#type' => 'hidden',
+      '#value' => _ding_nodelist_get_widget_human_name($widget),
+    );
+
+    $form['trim_content'][$widget]['title_trim'] = array(
+      '#type' => 'textfield',
+      '#size' => 3,
+      '#default_value' => variable_get('ding_nodelist_' . $widget . '_title_trim', NL_TITLE_SIZE),
```

Comment:
I expect trim values must be positive integers for this to work. Please add some validation e.g. `'#element_validate' => array('element_validate_integer_positive')`,